pkg/controller/perconaservermongodbrestore/physical.go

```diff
@@ -233,6 +233,23 @@ func (r *ReconcilePerconaServerMongoDBRestore) reconcilePhysicalRestore(ctx cont
 				return status, errors.Wrapf(err, "delete statefulset %s", stsName)
 			}
 		}
+
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			c := &psmdbv1.PerconaServerMongoDB{}
+			err := r.client.Get(ctx, types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, c)
+			if err != nil {
+				return err
+			}
+
+			orig := c.DeepCopy()
+			c.Annotations[psmdbv1.AnnotationResyncPBM] = "true"
```

Comment:
Maybe I'm missing something, but why do we need to do it like this with annotation, could we just perform the resync here directly (asynchronously)?

Reply:
statefulsets are deleted before this operation. we need to wait until cluster is ready, so we can run resync. that's why I'm doing this with annotation. for doing that here asynchronously, we need to check cluster status until it's ready and then run the resync in a separate goroutine. I think it'll be over complicated.